src/common/SmartNumericInputWithLabel/SmartNumericInputWithLabel.tsx

```diff
@@ -0,0 +1,53 @@
+import * as React from "react";
+import { SmartNumericInputProps, SmartNumericInput } from "../SmartNumericInput";
+
+export interface SmartNumericInputWithLabelProps extends SmartNumericInputProps {
+    label?: string;
+}
+
+export class SmartNumericInputWithLabel extends SmartNumericInput<SmartNumericInputWithLabelProps> {
```

Comment:
It doesn't work to use SmartNumericInputWithLabelProps?

Reply:
Not if I want to use the methods in SmartNumericInput also.

Reply:
You can probably extend the class of the component, and implement the interface for the property you need.

Reply:
If I do 
`export class SmartNumericInputWithLabel extends SmartNumericInput implements SmartNumericInputWithLabelProps {`

I'll get this error:
`
Property 'label' does not exist on type 'Readonly<SmartNumericInputProps> & Readonly<{ children?: ReactNode; }>'.
`
So the interface of SmartNumericInput is not a friend of interface SmartNumericInputWithLabelProps (even though the latter interface extends the former)

If I do 
`export class SmartNumericInputWithLabel extends SmartNumericInput implements React.Component<SmartNumericInputWithLabelProps, {}> {`
```
Class 'SmartNumericInputWithLabel' incorrectly implements class 'Component<SmartNumericInputWithLabelProps, {}, any>'. Did you mean to extend 'Component<SmartNumericInputWithLabelProps, {}, any>' and inherit its members as a subclass?
  Types of property 'props' are incompatible.
    Type 'Readonly<SmartNumericInputProps> & Readonly<{ children?: ReactNode; }>' is not assignable to type 'Readonly<SmartNumericInputWithLabelProps> & Readonly<{ children?: ReactNode; }>'.
      Type 'Readonly<SmartNumericInputProps> & Readonly<{ children?: ReactNode; }>' is not assignable to type 'Readonly<SmartNumericInputWithLabelProps>'.
        Type 'Readonly<SmartNumericInputProps> & Readonly<{ children?: ReactNode; }>' is missing the following properties from type 'Readonly<SmartNumericInputWithLabelProps>': context, setState, forceUpdate, render, and 3 more.
```